src/osd/OSD.cc

```diff
@@ -896,7 +898,7 @@ void OSDService::send_pg_temp()
   MOSDPGTemp *m = new MOSDPGTemp(osdmap->get_epoch());
   m->pg_temp = pg_temp_wanted;
   monc->send_mon_message(m);
-  _sent_pg_temp();
+  pg_temp_pending.swap(pg_temp_wanted);
```

Comment:
i don't think `pg_temp_pending.swap(pg_temp_wanted);` is equivalent to `pg_temp_pending.swap(pg_temp_wanted);` here. the former **moves** all elements from `pg_temp_wanted` into `pg_temp_pending`. and the latter **assigns** `pg_temp_pending` with `pg_temp_wanted`. and we can not have the splice operation for `std::map` until C++17, which offers `map::merge()`.